tests/cypress/integration/actions_tasks_objects/case_34_drawing_with_predefined_number_points.js

```diff
@@ -0,0 +1,113 @@
+
+import { taskName, labelName } from '../../support/const';
+
+context('Drawing with predefined number of points.', () => {
+    const caseId = '34';
+    const countPointsPolygon = 3;
+    const countPointsPolyline = 2;
+    const countPointsPoint = 1;
+
+    const createPolygonShape = {
+        reDraw: false,
+        type: 'Shape',
+        labelName: labelName,
+        pointsMap: [
+            { x: 500, y: 100 },
+            { x: 600, y: 100 },
+            { x: 500, y: 200 },
+        ],
+        numberOfPoints: 3,
+    };
+    const createPolylinesShape = {
+        type: 'Shape',
+        labelName: labelName,
+        pointsMap: [
+            { x: 500, y: 250 },
+            { x: 600, y: 250 },
+        ],
+        numberOfPoints: 2,
+    };
+    const createPointsShape = {
+        type: 'Shape',
+        labelName: labelName,
+        pointsMap: [{ x: 500, y: 200 }],
+        numberOfPoints: 1,
+    };
+
+    before(() => {
+        cy.openTaskJob(taskName);
+    });
+
+    function tryDrawObjectPredefinedNumberPoints(objest, pointsCount) {
+        cy.get(`.cvat-draw-${objest}-control`).click().wait(500);
+        cy.get('.cvat-draw-shape-popover')
+            .not('.ant-popover-hidden')
+            .within(() => {
+                cy.get('.cvat-draw-shape-popover-points-selector')
+                    .type(`${pointsCount - 1}`)
+                    .focused()
+                    .tab();
+                cy.get('[role="spinbutton"]').should('have.attr', 'aria-valuenow', pointsCount);
+            });
+    }
+
+    function tryDeletePoint() {
+        let svgJsCircleId = [];
+        cy.get('#cvat_canvas_shape_1').trigger('mousemove').should('have.attr', 'fill-opacity', 0.3);
+        cy.get('circle').then((circle) => {
+            for (let i = 0; i < circle.length; i++) {
+                if (circle[i].id.match(/^SvgjsCircle\d+$/)) {
+                    svgJsCircleId.push(circle[i].id);
+                }
+            }
+            cy.get(`#${svgJsCircleId[0]}`).click({ altKey: true });
+        });
+    }
+
+    function checkNotificationAndClose() {
+        cy.get('.cvat-notification-notice-update-annotations-failed')
+            .should('exist')
+            .within(() => {
+                cy.get('[aria-label="close"]').click();
+            });
+    }
+
+    describe(`Testing case "${caseId}"`, () => {
+        it('Start drawing a polygon. Set "Number of points" less then 3 and press Tab. "Number of points" takes the value 3 automatically.', () => {
+            tryDrawObjectPredefinedNumberPoints('polygon', countPointsPolygon);
+        });
+
+        it('Draw a polygon with 3 points. And try to delete one point. The error notification should appear.', () => {
+            cy.createPolygon(createPolygonShape);
+            tryDeletePoint();
+            checkNotificationAndClose();
+            cy.removeAnnotations(); // Removing the annotation for the convenience of further testing
```

Comment:
Just a question. As we are fighting for testing time is it faster to delete single object with for example delete key or remove all annotations. As i know, removing all annotation requests confirming action with popup dialog, so it consume some time. maybe it can be faster to clear all annotations after all test cases or delete single shape with delete key. What is your opinion?

Reply:
I tried the option of deleting the object. The time difference is insignificant. Also, the test fails when trying to activate the object due to its covered with the ```cvat_canvas_text_content``` element.